mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go

```diff
@@ -0,0 +1,193 @@
+	Expect(pvc).ToNot(BeNil())
+
+	// Wait for the PVC to be bound.
+	Eventually(func() coreV1.PersistentVolumeClaimPhase {
+		return Cmn.GetPvcStatusPhase(volName)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(coreV1.ClaimBound))
+
+	// Refresh the PVC contents, so that we can get the PV name.
+	pvc, getPvcErr = Cmn.GetPVC(volName)
+	Expect(getPvcErr).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
+
+	// Wait for the PV to be provisioned
+	Eventually(func() *coreV1.PersistentVolume {
+		pv, getPvErr := Cmn.GetPV(pvc.Spec.VolumeName)
+		if getPvErr != nil {
+			return nil
+		}
+		return pv
+
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Not(BeNil()))
+
+	// Wait for the PV to be bound.
+	Eventually(func() coreV1.PersistentVolumePhase {
+		return Cmn.GetPvStatusPhase(pvc.Spec.VolumeName)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(coreV1.VolumeBound))
+
+	// Wait for the MSV to be provisioned
+	Eventually(func() *Cmn.MayastorVolStatus {
+		return Cmn.GetMSV(string(pvc.ObjectMeta.UID))
+	},
+		defTimeoutSecs, //timeout
+		"1s",           // polling interval
+	).Should(Not(BeNil()))
+
+	// Wait for the MSV to be healthy
+	Eventually(func() string {
+		return Cmn.GetMsvState(string(pvc.ObjectMeta.UID))
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal("healthy"))
+
+	// Create the fio Pod
+	fioPodName := "fio-" + volName
+	pod, err := Cmn.CreateFioPod(fioPodName, volName)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pod).ToNot(BeNil())
+
+	// Wait for the fio Pod to transition to running
+	Eventually(func() bool {
+		return Cmn.IsPodRunning(fioPodName)
+	},
+		defTimeoutSecs,
+		"1s",
+	).Should(Equal(true))
+
+	// Run the fio test
+	Cmn.RunFio(fioPodName, 5)
+
+	// Delete the fio pod
+	err = Cmn.DeletePod(fioPodName)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Delete the PVC
+	deleteErr := Cmn.DeletePVC(volName)
+	Expect(deleteErr).To(BeNil())
+
+	// Wait for the PVC to be deleted.
+	Eventually(func() bool {
+		return Cmn.IsPVCDeleted(volName)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(true))
+
+	// Wait for the PV to be deleted.
+	Eventually(func() bool {
+		return Cmn.IsPVDeleted(pvc.Spec.VolumeName)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(true))
+
+	// Wait for the MSV to be deleted.
+	Eventually(func() bool {
+		return Cmn.IsMSVDeleted(string(pvc.ObjectMeta.UID))
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(true))
+}
+
+func stressTestPVC() {
+	for ix := 0; ix < 10; ix++ {
+		testPVC(fmt.Sprintf("stress-pvc-nvmf-%d", ix), "mayastor-nvmf")
+		testPVC(fmt.Sprintf("stress-pvc-iscsi-%d", ix), "mayastor-iscsi")
+		// FIXME: Without this delay GetPvcClaimStatusPhase returns Pending
+		// even though kubectl shows that the pvc is Bound.
+		//pause()
```

Comment:
Can this be removed as the delay isn't used?

Reply:
Done